### PR TITLE
Integrate GolangCi-Lint 

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
I tried breaking up the changes into individual commits. But overall, the changes are a bit of a mess. Please point out any shady code pieces you find.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-mattermod/issues/131

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -35,23 +35,19 @@ type Store interface {
 }
 
 type PullRequestStore interface {
-	Save(pr *model.PullRequest) StoreChannel
-	Get(repoOwner, repoName string, number int) StoreChannel
-	List() StoreChannel
```
Comment (hanzei) [inline store/store.go:40]: I did drop this method given that it's not longer used

File: server/spinmint.go [hunk 2]
```diff
@@ -157,7 +68,9 @@ func (s *Server) waitForBuildAndSetupSpinmint(pr *model.PullRequest, upgradeServ
 		s.storeSpinmintInfo(spinmint)
 	} else {
 		spinmint := result.Data.(*model.Spinmint)
-		instance.InstanceId = aws.String(spinmint.InstanceId)
+		instance = &ec2.Instance{
+			InstanceId: aws.String(spinmint.InstanceID),
+		}
```
Comment (hanzei) [inline server/spinmint.go:73]: Fixed an NPE here

**Comment by hanzei** [comment]
@metanerd Given that there where a lot of changes, please tripple check that nothing is broken.

**Comment by metanerd** [comment]
I will deploy and monitor.

**Comment by metanerd** [comment]
Looks very good so far! :heart:

**Pull request merged**